src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPluginWithCompute.java

```diff
@@ -14,6 +24,12 @@
 
   // returns the compute-spec class this visualization expects
   protected abstract Class<R> getComputeSpecClass();
+  protected List<StreamSpec> getRequestedStreams(S pluginSpec) {
+    return getRequestedStreams(pluginSpec, _computeSpec);
+  }
+  protected abstract List<StreamSpec> getRequestedStreams(S pluginSpec, R computeConfig);
+
+  protected static final String COMPUTE_STREAM_NAME = "compute_input_dataset";
```

Comment:
My impression was that we would somehow tell the merging service how to fetch the compute result stream for possible merging with regular and derived vars rather than ask compute service for it directly.  Is that a wrong understanding?

Reply:
Oh, I see below- we are still getting them from merging in a second stream.  Ok if the subset is still applied here?

Reply:
Yea. I think it's fine the filtering is applied. 

Right now we make two requests, one is input to the compute, the other just raw inputs for the viz. And the viz plugin actually merges the output of the compute to the input for the viz. 

In the ideal, the viz plugin makes a single request to merging, and the compute output and other viz inputs are merged there like you described. And the compute service would make its own request to merging for its own inputs, w the same filtering criteria applied.